result_macros.h raises error if NTSTATUS values are defined as NTSTATUS types

NTSTATUS values are defined as NTSTATUS types (implicitly, via some combination of Windows headers) in our project, leading to a [recent WIL change](https://github.com/microsoft/wil/pull/553/files#diff-0989dbf1a874b464dff301ca2ce90faeeee720b8412a2d2c37b66a4a4cdc9fc2L4522-R4541) introducing build-breaking errors:

```
submodules\wil\include\wil\result_macros.h(4541,29): warning C4245: '=': conversion from 'NTSTATUS' to 'D
       WORD', signed/unsigned mismatch
```

I'm not a C++ expert, but I think the original cast was required if the type of the `STATUS_XXX` definitions vary by environment.

--- a/include/wil/result_macros.h
+++ b/include/wil/result_macros.h
@@ -4537,8 +4537,9 @@ namespace details
 
         // parameter 0 is the !analyze code (FAST_FAIL_FATAL_APP_EXIT)
         EXCEPTION_RECORD exr{};
-        exr.NumberParameters = 1;                        // default to be safe, see below
-        exr.ExceptionCode = STATUS_STACK_BUFFER_OVERRUN; // 0xC0000409
+        exr.NumberParameters = 1; // default to be safe, see below
+        // NOLINTNEXTLINE(readability-redundant-casting): STATUS_ macros inconsistently defined in Windows headers
+        exr.ExceptionCode = static_cast<DWORD>(STATUS_STACK_BUFFER_OVERRUN); // 0xC0000409
         exr.ExceptionFlags = EXCEPTION_NONCONTINUABLE;
         exr.ExceptionInformation[0] = FAST_FAIL_FATAL_APP_EXIT; // see winnt.h, generated from minkernel\published\base\ntrtl_x.w
         if (failure.returnAddress == nullptr) // FailureInfo does not have _ReturnAddress, have RaiseFailFastException generate it

